libs/prisma-models/src/model.rs

```diff
@@ -84,35 +87,17 @@ impl Model {
     /// The identifier is nothing but an internal convention to have an anchor point for querying, or in other words,
     /// the identifier is not to be mistaken for a stable, external identifier, but has to be understood as
     /// implementation detail that is used to reason over a fixed set of fields.
-    ///
-    /// The rules for determining the primary identifier are as follows:
-    /// 1. If an ID definition (single or multi-part doesn't matter) is present, take that one.
-    /// 2. If no ID definition is found, take the first scalar unique found that is required.
-    /// 3. If no scalar unique is found, take the first compound unique found. All fields must be required.
-    /// 4. If all of the above fails, we panic. Models with no unique / ID are not supported (yet).
     pub fn primary_identifier(&self) -> ModelProjection {
-        let fields: Vec<_> = self
-            .fields()
-            .id()
-            .or_else(|| {
-                self.fields()
-                    .scalar()
-                    .into_iter()
-                    .find(|sf| sf.is_unique && sf.is_required)
-                    .map(|x| vec![x])
-            })
-            .or_else(|| {
-                self.unique_indexes()
-                    .into_iter()
-                    .find(|index| index.fields().into_iter().all(|f| f.is_required))
-                    .map(|index| index.fields().into_iter().map(|f| f.into()).collect())
+        let dml_fields = self.dml_model.first_unique_criterion();
```

Comment:
I'm not a fan of this approach, because it means that on every call to `primary_identifier` we go through the entire call chain to determine the identifier. This was okay before as we didn't have the DML precomputing things and I didn't want to cache the identifier in the first version of this function, but with the code factored out, I don't see why we wouldn't just precompute the criterias and store them on the internal model, or even just the primary ID.